hardware_interface/include/hardware_interface/hardware_info.hpp

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020 ros2_control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#ifndef HARDWARE_INTERFACE__HARDWARE_INFO_HPP_
+#define HARDWARE_INTERFACE__HARDWARE_INFO_HPP_
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace hardware_interface
+{
+
+/**
+ * \brief This structure stores information about hardware defined in a robot's URDF.
```

Comment:
same comment here, this has to be made a bit more verbose for people.

Reply:
see above...